### PR TITLE
[MIG] auditlog: Migration to 12.0

### PR DESCRIPTION
**Pull request by sebalix**
Supersedes #1392 

I know several PR were opened to migrate this module, and the existing one https://github.com/OCA/server-tools/pull/1392 broke the history of the repository.

This PR also set all rule fields as readonly when the rule is subscribded.

### CODE REVIEW

File: auditlog/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# © 2015 ABF OSIELL <https://osiell.com>
```
Comment (simahawk) [inline auditlog/__manifest__.py]: Copyright

**Review by simahawk — APPROVED** [review]

File: auditlog/data/ir_cron.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+
+        <record id="ir_cron_auditlog_autovacuum" model="ir.cron">
+            <field name='name'>Auto-vacuum audit logs</field>
+            <field name='interval_number'>1</field>
+            <field name='interval_type'>days</field>
+            <field name="numbercall">-1</field>
```
Comment (Tonow-c2c) [inline auditlog/data/ir_cron.xml:8]: Just to know why `-1` ?

File: auditlog/models/http_session.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from odoo.http import request
+
+
+class AuditlogtHTTPSession(models.Model):
+    _name = 'auditlog.http.session'
+    _description = "Auditlog - HTTP User session log"
+    _order = "create_date DESC"
+
+    display_name = fields.Char(
+        "Name", compute="_compute_display_name", store=True)
```
Comment (Tonow-c2c) [inline auditlog/models/http_session.py:14]: I prefer in multi line but it's ok like this

**Review by Tonow-c2c — APPROVED** [review]
LGTM

**Review by chienandalu — APPROVED** [review]
Maybe some commits could be squashed to reduce history. Anyway :+1:

**Comment by sebalix** [comment]
@chienandalu I squashed translations ones the best I could, about the others I don't know if it's easy (to keep authorship especially).

**Comment by chienandalu** [comment]
Well, yeah, I meant consecutive commits of the same author :)

**Pull request merged**